Added openfst package

mcernak opened the pull request:
Added conda recipe for the latest OpenFst Version 1.6.1.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/openfst```) and found some lint.

Here's what I've got...


For **recipes/openfst**:

 * The recipe must have some tests.
 * Selectors are suggested to take a ``<two spaces>#<one space>[<expression>]`` form.
 * There are 1 too many lines.  There should be one empty line at the end of the file.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/openfst```) and found it was in an excellent condition.

183amir commented on recipes/openfst/meta.yaml:
```diff
@@ -0,0 +1,33 @@
+
+requirements:
+  build:
+    - gcc >=4.7  # [linux]
```
remove this.

183amir replied:
Add `toolchain`

183amir commented on recipes/openfst/meta.yaml:
```diff
@@ -0,0 +1,33 @@
+    - gcc >=4.7  # [linux]
+    - python
+  run:
+    - gcc >=4.7  # [linux]
```
remove this.

183amir commented on recipes/openfst/build.sh:
```diff
@@ -0,0 +1,9 @@
+if [ "`uname`" != "Darwin" ]; then
+  export CC=${PREFIX}/bin/gcc
+  export CXX=${PREFIX}/bin/g++
+fi
```
remove this.

183amir commented on recipes/openfst/build.sh:
```diff
@@ -0,0 +1,9 @@
+  export CC=${PREFIX}/bin/gcc
+  export CXX=${PREFIX}/bin/g++
+fi
+
```
`chmod +x ./configure`

183amir commented on recipes/openfst/build.sh:
```diff
@@ -0,0 +1,9 @@
+  export CXX=${PREFIX}/bin/g++
+fi
+
+./configure --prefix=${PREFIX} || exit 1
```
remove `|| exit 1`

183amir commented on recipes/openfst/meta.yaml:
```diff
@@ -0,0 +1,33 @@
+package:
+  name: openfst
+  version: "0.1"
```
Put openfst version.

183amir commented:
Now make sure the CI is green. Ask questions if you can't fix them.

mcernak commented:
I added toolchain requirement, and the build now fails due to aclocal 
dependency

WARNING: 'aclocal-1.14' is missing on your system. You should only need 
it if you modified 'acinclude.m4' or 'configure.ac' or m4 files included 
by 'configure.ac'. The 'aclocal' program is part of the GNU Automake 
package:

So I added also the automake requirement, however, the error is not 
resolved. I try to fix it.

Milos

On 10.03.2017 11:11, Amir Mohammadi wrote:
>
> Now make sure the CI is green. Ask questions if you can't fix them.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/conda-forge/staged-recipes/pull/2569#issuecomment-285630291>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AEcac8tMUWJY7y3kPnf_l-Su3j2PQ-SKks5rkSHtgaJpZM4MWk_D>.
>

183amir commented:
Maybe you need `autoconf`

183amir commented on recipes/openfst/meta.yaml:
```diff
@@ -0,0 +1,38 @@
+  build:
+    - toolchain
+    - automake
+    - python
```
Is `python` needed?

183amir commented on recipes/openfst/meta.yaml:
```diff
@@ -0,0 +1,38 @@
+    - autoconf
+    - python
+  run:
+    - toolchain
```
`toolchain` should not be a run requirement.

183amir commented on recipes/openfst/meta.yaml:
```diff
@@ -0,0 +1,38 @@
+
+test:
+  requires:
+    - toolchain
```
Can you remove it from test for now? If the tests fail we might want to put it back.

183amir commented:
You might want to take a look at here too: https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=openfst

183amir commented:
Myabe you need to run `./configure.ac` first?

mcernak commented:
The CircleCI test fails with build dependencies. Maybe the test uses 
automake 1.15 (the last version), while OpenFST requires 1.14. I tried 
to specify it in the meta file but it did not help..

Milos


On 10.03.2017 11:41, Amir Mohammadi wrote:
>
> Myabe you need to run |./configure.ac| first?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/conda-forge/staged-recipes/pull/2569#issuecomment-285636622>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AEcaczhHCma52DULkRwbbsfy430h2pXoks5rkSjigaJpZM4MWk_D>.
>

183amir commented:
Like I said, can you run ./configure.a.c before ./configure?

mcernak commented:
./configure.ac is not a script, it is used by autoconf to create 
./configure:

https://www.gnu.org/software/autoconf/manual/autoconf-2.68/html_node/autoconf-Invocation.html

This is what I call in build.sh


On 11.03.2017 22:58, Amir Mohammadi wrote:
>
> Like I said, can you run ./configure.a.c before ./configure?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/conda-forge/staged-recipes/pull/2569#issuecomment-285903303>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AEcacz5zHIBeyW8DtcxAjbtwPe_-G5gUks5rkxkPgaJpZM4MWk_D>.
>
